libraries/AP_HAL_ChibiOS/hwdef/GEPRCF745BTHD/hwdef.dat

```diff
@@ -0,0 +1,185 @@
+# hw definition file for processing by chibios_pins.py
+# for GEPRCF745-BT-HD hardware.
+# thanks to betaflight for pin information
+
+# MCU class and specific type
+MCU STM32F7xx STM32F745xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1500
+
+# crystal frequency, setup to use external oscillator
+OSCILLATOR_HZ 8000000
+
+FLASH_SIZE_KB 1024
+
+# leave 2 sectors free
+FLASH_RESERVE_START_KB 96
+
+
+# only one I2C bus
+I2C_ORDER I2C1
+
+# I2C1 for baro
+PB8 I2C1_SCL I2C1
+PB9 I2C1_SDA I2C1
+
+# order of UARTs (and USB), 
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 EMPTY USART6 UART7 UART8
+
+# buzzer
+#PD2 TIM3_ETR TIM3 GPIO(77) ALARM
+#PD2 BUZZER OUTPUT GPIO(80) LOW
+#define HAL_BUZZER_PIN 80
+#
+
+# PA10 IO-debug-console
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# SPI1 for MPU6000
+PA4 MPU6000_CS CS
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+
+# SPI2 for ICM42688
+PB12 ICM42605_CS CS
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+# SPI3 for SDCard
+PA15 SDCARD_CS CS
+PC10 SPI3_SCK SPI3
+PC11 SPI3_MISO SPI3
+PC12 SPI3_MOSI SPI3
+
+# SPI4 for MAX7456 OSD
+PE4 MAX7456_CS CS
+PE2 SPI4_SCK SPI4
+PE5 SPI4_MISO SPI4
+PE6 SPI4_MOSI SPI4
+
+PC3 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC2 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+# define default battery setup
+define HAL_BATT_MONITOR_DEFAULT 4
+define HAL_BATT_VOLT_PIN 13
+define HAL_BATT_CURR_PIN 12
+define HAL_BATT_VOLT_SCALE 10.9
+define HAL_BATT_CURR_SCALE 28.5
+
+PC5 RSSI_ADC ADC1
+define BOARD_RSSI_ANA_PIN 15
+
+PC13 LED0 OUTPUT LOW GPIO(1) # LED
+define HAL_GPIO_A_LED_PIN 1
+define HAL_GPIO_B_LED_PIN 2
+define HAL_GPIO_LED_OFF 1
+
+# In order to accommodate bi-directional dshot certain devices cannot be DMA enabled
+# NODMA indicates these devices, if you remove it they will still not be resolved for DMA
+
+# USART1
+PA10 USART1_RX USART1
+PA9  USART1_TX USART1
+
+# USART2
+# RC input defaults to UART to allow for bi-dir dshot
+PA2 USART2_TX USART2
+PA3 USART2_RX USART2
+define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_RCIN
+define DEFAULT_SERIAL2_BAUD 115
+
+# USART3 (BT)
+PB11 USART3_RX USART3
+PB10 USART3_TX USART3
+
+# UART4
+PA0 UART4_TX UART4
+PA1 UART4_RX UART4
+
+# UART6
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+
+# UART7
+PE7 UART7_RX UART7
+PE8 UART7_TX UART7
+
+# UART8
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+
+# Motors, bi-directional dshot capable
+PB0     TIM3_CH3 TIM3 PWM(1) GPIO(50)           # M1
```

Comment:
You will need to add BIDIR to some of the outputs to make them bi-directional capable. This might be tricky on an F7 due to DMA sharing

Reply:
We are not quite sure about the Settings of BIDIR and DMA, and may need your assistance. If you need to get FC samples for debugging, we can send them by mail.

Reply:
I've had a go - but maybe send me a sample anyway?